src/Wallabag/CoreBundle/Controller/TagController.php

```diff
@@ -152,27 +152,40 @@ public function renameTagAction(Tag $tag, Request $request)
         $form->handleRequest($request);
 
         if ($form->isSubmitted() && $form->isValid()) {
-            $entries = $this->get('wallabag_core.entry_repository')->findAllByTagId(
-                $this->getUser()->getId(),
-                $tag->getId()
-            );
-            foreach ($entries as $entry) {
-                $this->get('wallabag_core.tags_assigner')->assignTagsToEntry(
-                    $entry,
-                    $form->get('label')->getData()
+            $formLabel = $form->get('label')->getData();
+            $newTag = new Tag();
+            $newTag->setLabel($formLabel);
+
+            if ($newTag->getLabel() !== $tag->getLabel()) {
```

Comment:
I tend to avoid using condition which generate too much code indented below.

Maybe you can move the redirect url definition above that condition, flip the check to have a small part in the `if` and return early:

```php
$redirectUrl = $this->get('wallabag_core.helper.redirect')->to($request->headers->get('referer'), '', true);

if ($newTag->getLabel() === $tag->getLabel()) {
    return $this->redirect($redirectUrl);
}

// whole code about renaming

return $this->redirect($redirectUrl);
```